openai/api_resources/image.py

```diff
@@ -31,12 +34,20 @@ def create(
             organization=organization,
         )
 
-        _, api_version = cls._get_api_type_and_version(api_type, api_version)
+        api_type, api_version = cls._get_api_type_and_version(api_type, api_version)
 
         response, _, api_key = requestor.request(
-            "post", cls._get_url("generations"), params
+            "post", cls._get_url("generations", azure_action="submit", api_type=api_type, api_version=api_version), params
         )
 
+        if api_type in (util.ApiType.AZURE, util.ApiType.AZURE_AD):
+            requestor.api_base = "" # operation_location is a full url
+            response, _, api_key = requestor.poll(
+                "get", response.operation_location,
+                until=lambda response: response.data["status"] not in ["notRunning", "running"],
+                failed=lambda response: response.data['status'] in [ 'canceled', 'failed']
```

Comment:
Does 'canceled' mean canceled by the service? Or is there a way a user can cancel an operation?

Reply:
removed canceled from the check.